render/naga-agal/src/builder.rs

```diff
@@ -1060,45 +1097,54 @@ impl<'a> NagaBuilder<'a> {
                 });
                 self.emit_dest_store(dest, expr)?;
             }
-            // Perform 'M * v', where M is a 4x4 matrix, and 'v' is a column vector.
-            Opcode::M44 => {
+            // Perform 'M * v', where M is a matrix, and 'v' is a column vector.
+            Opcode::M33 | Opcode::M34 | Opcode::M44 => {
                 let source2 = match source2 {
                     Source2::SourceField(source2) => source2,
                     _ => unreachable!(),
                 };
 
+                let (num_rows, ty, vec_size, out_size) = match opcode {
+                    Opcode::M33 => (
+                        3,
+                        self.matrix3x3f,
+                        VectorSize::Tri,
+                        VertexAttributeFormat::Float3,
+                    ),
+                    Opcode::M34 => (
+                        3,
```

Comment:
I do not understand the code or this area of the codebase or know about NAGA or AGAL, but, is it correct that this value is `3` instead of `4`?

https://github.com/ruffle-rs/ruffle/pull/10589/files?diff=unified&w=1#diff-de78bf7c3d3c1d8988c51e39816e4dd5d4bd731f862ce46da4afb97342464c1aR376 seems, at a superficial glance at least, like it maybe should be 4 - unless rows and columns are counted differently, maybe. But I know very little about this.